### PR TITLE
sample-gltf-viewer: introduce QR code feature.

### PR DESCRIPTION
**Pull request by prideout**
This works in tandem with the new page at https://google.github.io/filament/qrcode/

Screen Shot 2021-03-01 at 9 46 36 AM

This feature did not require any changes to `ModelViewer.kt` or any of our libraries. It involves the following additions to the app MainActivity:

- Add a button to the app bar, which requires AppCompatActivity as per
  https://developer.android.com/training/appbar/setting-up

- Add a PreviewView widget from androidx.

- Add usage of BarcodeScanner in mlkit.

- Add a coroutine for downloading data via HttpURLConnection.

We would like to add more features in the future, including the ability
to drop additional models without rescanning and the ability to drop
zip files.

This feature was inspired by a Model Viewer editor feature, thanks @elalish

### CODE REVIEW

**Comment by elalish** [comment]
Let me get this straight: you also created a project called `ModelViewer`? 🤦‍♂️

**Comment by prideout** [comment]
> Let me get this straight: you also created a project called `ModelViewer`? 🤦‍♂️

Well it's just the name of a Kotlin class, it seemed like the most appropriate name at the time.  :)

File: android/samples/sample-gltf-viewer/src/main/java/com/google/android/filament/gltf/MainActivity.kt [hunk 1]
```diff
@@ -74,6 +111,126 @@ class MainActivity : Activity() {
         modelViewer.view.bloomOptions = bloomOptions
     }
 
+    private fun bindPreview(cameraProvider: ProcessCameraProvider) {
+        val preview : Preview = Preview.Builder().build().also {
```
Comment (bejado) [inline android/samples/sample-gltf-viewer/src/main/java/com/google/android/filament/gltf/MainActivity.kt:115]: I think you can leave off the `: Preview` here and `: CameraSelector` down below

File: android/samples/sample-gltf-viewer/src/main/java/com/google/android/filament/gltf/MainActivity.kt [hunk 2]
```diff
@@ -74,6 +111,126 @@ class MainActivity : Activity() {
...
+        if (previewView.visibility != View.VISIBLE) {
+            imageProxy.close()
+            return
+        }
+        val options = BarcodeScannerOptions.Builder().setBarcodeFormats(Barcode.FORMAT_QR_CODE).build()
+        val image = InputImage.fromMediaImage(mediaImage, imageProxy.imageInfo.rotationDegrees)
+        val scanner = BarcodeScanning.getClient(options)
+        scanner.process(image)
+                .addOnSuccessListener { barcodes ->
+                    for (barcode in barcodes) {
```
Comment (bejado) [inline android/samples/sample-gltf-viewer/src/main/java/com/google/android/filament/gltf/MainActivity.kt:173]: If multiple barcodes are detected, does it make sense to call `downloadFromBarcode` for each? Perhaps we just want to download the first one we find.

**Review by bejado — APPROVED** [review]

File: android/samples/sample-gltf-viewer/src/main/java/com/google/android/filament/gltf/MainActivity.kt (hunk 2 above)
Comment (prideout) [reply android/samples/sample-gltf-viewer/src/main/java/com/google/android/filament/gltf/MainActivity.kt:173]: Good catch, I will simplify this.

**Pull request closed without merging**